wapitiCore/attack/mod_csp.py

```diff
@@ -0,0 +1,77 @@
+import requests, re
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import Additional, _
+
+
+# This module check the basics recommendations of CSP
+class mod_csp(Attack):
+    name = "csp"
+
+    checks = ["default-src", "script-src", "object-src", "base-uri"]
+
+    check_default = ["default-src *", "default-src 'unsafe-eval'"]
+    check_script_src = ["script-src 'unsafe-inline'", "script-src 'data:'", "script-src 'http:'", "script-src 'https:'"]
```

Comment:
Using the policy prefix  each time you won't match values when they aren't in the first position.

Such an example is shown here :
https://developers.google.com/web/fundamentals/security/csp/#use_case_3_ssl_only

`script-src https: 'unsafe-inline';`

Here `unsafe-inline` is in second position but it won't be detected.

Also it seems that `http:` or `https:` or `https://*.example.com` or `http://example.com` doesn't need to be put between quotes.

It should be possible to use a regex (or a very basic parser) to get the several values of the policy.

Reply:
Here is an example : 

```
import re

regex = re.compile(r"\s*((?:'[^']*')|(?:[^'\s]+))\s*")
csp = "default-src https: https://cdn.example.net; script-src https: 'unsafe-inline' 'nonce-EDNnf03nceIOfn39fn3e9h3sdfa' 'sha256-qznLcsROx4GACP2dm0UCKCzCG-HiZ1guq6ZZDob_Tng='; style-src https: 'unsafe-inline'"

for policy_string in csp.split(";"):
    policy_name, policy_values = policy_string.strip().split(" ", 1)
    print(policy_name)
    print("="*20)
    for value in regex.findall(policy_values):
        print(value.strip("'"))
    print('')
```

output : 

```
default-src
====================
https:
https://cdn.example.net

script-src
====================
https:
unsafe-inline
nonce-EDNnf03nceIOfn39fn3e9h3sdfa
sha256-qznLcsROx4GACP2dm0UCKCzCG-HiZ1guq6ZZDob_Tng=

style-src
====================
https:
unsafe-inline
```